bindings.c

```diff
@@ -4092,6 +4292,363 @@ static int proc_swaps_read(char *buf, size_t size, off_t offset,
+ */
+static int refresh_load(struct load_node *p, char *path)
+{
+	FILE *f = NULL;
+	char **idbuf;
+	char proc_path[256];
+	int i, ret, run_pid = 0, total_pid = 0, last_pid = 0;
+	char *line = NULL;
+	size_t linelen = 0;
+	int sum, length;
+	DIR *dp;
+	struct dirent *file;
+
+	idbuf = (char **)malloc(sizeof(char *));
+	if (!idbuf) {
+		lxcfs_error("Refresh container:%s failed for malloc!\n", p->cg);
+		return -1;
+	}
+	sum = calc_pid(&idbuf, path, DEPTH_DIR, 0, p->cfd);
+	/*  normal exit and abnormal exit */
+	if (sum == 0 || sum == -1)
+		goto out;
+
+	for (i = 0; i < sum; i++) {
+		/*clean up '\n' */
+		length = strlen(idbuf[i])-1;
+		idbuf[i][length] = '\0';
+		do {
+			ret = snprintf(proc_path, 256, "/proc/%s/task", idbuf[i]);
+		} while (ret < 0);
+
+		dp = opendir(proc_path);
+		if (!dp) {
+			lxcfs_error("%s\n", "Open proc_path failed in refresh_load.");
+			continue;
+		}
+		while ((file = readdir(dp)) != NULL) {
+			if (strncmp(file->d_name, ".", 1) == 0)
+				continue;
+			total_pid++;
+			/* We make the biggest pid become last_pid.*/
+			ret = atof(file->d_name);
+			last_pid = (ret > last_pid) ? ret : last_pid;
+
+			do {
+				ret = snprintf(proc_path, 256, "/proc/%s/task/%s/status", idbuf[i], file->d_name);
+			} while (ret < 0);
+			f = fopen(proc_path, "r");
+			if (f != NULL) {
+				while (getline(&line, &linelen, f) != -1) {
+					/* Find State */
+					if ((line[0] == 'S') && (line[1] == 't'))
+						break;
+				}
+			if ((line[7] == 'R') || (line[7] == 'D'))
+				run_pid++;
+			fclose(f);
+			}
+		}
+		closedir(dp);
+	}
+	/*Calculate the loadavg.*/
+	p->avenrun[0] = calc_load(p->avenrun[0], EXP_1, run_pid);
+	p->avenrun[1] = calc_load(p->avenrun[1], EXP_5, run_pid);
+	p->avenrun[2] = calc_load(p->avenrun[2], EXP_15, run_pid);
+	p->run_pid = run_pid;
+	p->total_pid = total_pid;
+	p->last_pid = last_pid;
+
+	free(line);
+	for (; i > 0; i--)
+		free(idbuf[i-1]);
+out:
+	free(idbuf);
+	return sum;
+}
+/*
+ * Traverse the hash table and update it.
+ */
+void *load_begin(void *arg)
+{
+
+	char *path = NULL;
+	int i, sum, length, ret;
+	struct load_node *f;
+	int first_node;
+	clock_t time1, time2;
+
+	while (1) {
+		time1 = clock();
+		for (i = 0; i < LOAD_SIZE; i++) {
+			pthread_mutex_lock(&load_hash[i].lock);
+			if (load_hash[i].next == NULL) {
+				pthread_mutex_unlock(&load_hash[i].lock);
+				continue;
+			}
+			f = load_hash[i].next;
+			first_node = 1;
+			while (f) {
+				length = strlen(f->cg) + 2;
+				do {
+					/* strlen(f->cg) + '.' or '' + \0 */
+					path = (char *)malloc(length);
+				} while (!path);
+
+				ret = snprintf(path, length, "%s%s", *(f->cg) == '/' ? "." : "", f->cg);
+				if (ret < 0 || ret > length - 1) {
+					/* snprintf failed, ignore the node.*/
+					lxcfs_error("Refresh node %s failed for snprintf().\n", f->cg);
+					goto out;
+				}
+				sum = refresh_load(f, path);
+				if (sum == 0) {
+					f = del_node(f, i);
+				} else {
+out:				f = f->next;
```

Comment:
Wrong indentation.

Reply:
The format on the local is correct, but I don't know why it was wrong after pushing.
I will try to solve it.